packages/tss/src/tss.ts

```diff
@@ -0,0 +1,811 @@
+      const pubKeyPoint = tssCommits[0].toEllipticPoint(ec);
+      const devicePubKeyPoint = pubKeyPoint.add(noncePub);
+      return Point.fromElliptic(devicePubKeyPoint);
+    }
+    return tssCommits[0];
+  }
+
+  /**
+   * Returns the node details for RSS.
+   */
+  async _getRssNodeDetails(): Promise<{
+    serverEndpoints: string[];
+    serverPubKeys: PointHex[];
+    serverThreshold: number;
+  }> {
+    const { serverEndpoints, serverPubKeys, serverThreshold } = await this.serviceProvider.getRSSNodeDetails();
+    if (!Array.isArray(serverEndpoints) || serverEndpoints.length === 0) throw new Error("service provider tss server endpoints are missing");
+    if (!Array.isArray(serverPubKeys) || serverPubKeys.length === 0) throw new Error("service provider pub keys are missing");
+    return {
+      serverEndpoints,
+      serverPubKeys,
+      serverThreshold: serverThreshold || Math.floor(serverEndpoints.length / 2) + 1,
+    };
+  }
+
+  /**
+   * Imports an existing private key for threshold signing. A corresponding user
+   * key share will be stored under the specified factor key.
+   */
+  async importTssKey(
+    params: {
+      tag: string;
+      importKey: Buffer;
+      factorPub: Point;
+      newTSSIndex: number;
+    },
+    serverOpts: {
+      selectedServers?: number[];
+      authSignatures: string[];
+    }
+  ): Promise<void> {
+    const ec = this._tssCurve;
+    if (!this.privKey) throw CoreError.privateKeyUnavailable();
+    if (!this.metadata) {
+      throw CoreError.metadataUndefined();
+    }
+    const { importKey, factorPub, newTSSIndex, tag } = params;
+
+    const oldTag = this.tssTag;
+    this._tssTag = tag;
+
+    try {
+      const { selectedServers = [], authSignatures = [] } = serverOpts || {};
+
+      if (!tag) throw CoreError.default(`invalid param, tag is required`);
+      if (!factorPub) throw CoreError.default(`invalid param, newFactorPub is required`);
+      if (!newTSSIndex) throw CoreError.default(`invalid param, newTSSIndex is required`);
+      if (authSignatures.length === 0) throw CoreError.default(`invalid param, authSignatures is required`);
+
+      const existingFactorPubs = this.metadata.factorPubs[tag];
+      if (existingFactorPubs?.length > 0) {
+        throw CoreError.default(`Duplicate account tag, please use a unique tag for importing key`);
+      }
+      const factorPubs = [factorPub];
+
+      const importScalar = await (async () => {
+        if (this._tssKeyType === KeyType.secp256k1) {
+          return new BN(importKey);
+        } else if (this._tssKeyType === KeyType.ed25519) {
+          // Store seed in metadata.
+          const domainKey = getEd25519SeedStoreDomainKey(this.tssTag || TSS_TAG_DEFAULT);
+          const result = this.metadata.getGeneralStoreDomain(domainKey) as Record<string, unknown>;
+          if (result) {
+            throw new Error("Seed already exists");
+          }
+
+          const { scalar } = getEd25519ExtendedPublicKey(importKey);
```

Comment:
function name should be renamed to getEd25519KeyPairFromSeed ?

Reply:
done